yt/data_objects/construction_data_containers.py

```diff
@@ -279,18 +279,23 @@ def get_data(self, fields=None):
         # TODO: Add the combine operation
         xax = self.ds.coordinates.x_axis[self.axis]
         yax = self.ds.coordinates.y_axis[self.axis]
-        ox = self.ds.domain_left_edge[xax].v
-        oy = self.ds.domain_left_edge[yax].v
-        px, py, pdx, pdy, nvals, nwvals = tree.get_all(False, merge_style)
+
+        ix, iy, ires, nvals, nwvals = tree.get_all(False, merge_style)
+        px, pdx = self.ds.index._icoords_to_fcoords(
+            ix[:, None], ires // self.ds.ires_factor, axes=(xax,)
```

Comment:
In testing with yt_idefix, I found a problem with this line:
since `ires_factor` is defined as `log2(ds.refine_by)`, this leads to a `RuntimeWarning: divide by zero encountered in floor_divide` for Dataset subclasses that set `refine_by = 1` (usually meaning "this isn't AMR, leave me be"). We can solve this by
- special casing for `self.ds.ires_factor == 0` here
- modify `ires_factor` somehow
- set a standard for what to do with `refine_by` in formats that are not AMR (set `refine_by = 2` anyway ? override the `ires_factor` property ?)

It's definetly a bug, I'm just not sure what part, this branch or frontends, need a fix

Reply:
Hmm, good point.  My *first* inclination is to not allow `ires_factor` to go below 1.  The point really is to make a number that we can use here, and not really anywhere else (yet).

Reply:
Sounds good. Now that I think about it, `refine_by = 1` doesn't make much sense (the real issue being that this attribute is needed even for formats where it's irrelevant -> #1880)
Maybe it would also be ok to change these rare frontends to use `refine_by = 2` instead ? this would avoid increasing the complexity of the base implementation of `ires_factor`. WDYT ?

Reply:
It would, but I think the change to ires factor is fine as well.

Reply:
sure, we can absolutely do both

Reply:
or not, being able to grep "refine_by = 1" has value in a sense.